pinot-core/src/main/java/com/linkedin/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -92,6 +95,8 @@ public PlanNode makeInnerSegmentPlan(IndexSegment indexSegment, BrokerRequest br
           // Aggregation only query.
         if (isFitForMetadataBasedPlan(brokerRequest)) {
           return new MetadataBasedAggregationPlanNode(indexSegment, brokerRequest.getAggregationsInfo());
+        } else if (isFitForDictionaryBasedPlan(brokerRequest, indexSegment)) {
```

Comment:
Not a blocker for this PR, but seems like the number of if's are increasing. May be better to have a factory that returns the correct plan node.